lib/libutee/tee_api_arith_mpi.c

```diff
@@ -443,7 +443,25 @@ void TEE_BigIntNeg(TEE_BigInt *dest, const TEE_BigInt *src)
 void TEE_BigIntMul(TEE_BigInt *dest, const TEE_BigInt *op1,
 		   const TEE_BigInt *op2)
 {
-	bigint_binary(dest, op1, op2, mbedtls_mpi_mul_mpi);
+	size_t bs1 = TEE_BigIntGetBitCount(op1);
+	size_t bs2 = TEE_BigIntGetBitCount(op2);
+	size_t s = TEE_BigIntSizeInU32(bs1) + TEE_BigIntSizeInU32(bs2);
```

Comment:
I think we're overshooting with `2` here, considering how `TEE_BigIntSizeInU32()` is defined, but that quite harmless so I guess we can leave it as it is.

Reply:
I think you're right, would `TEE_BigIntSizeInU32(bs1 + bs2)` be more correct? I could not convince myself it is certain to be enough, so the current form may be the most correct way, short of subtracting 2 (which would be hackish, relying on the internals of the macro).

Reply:
I'm not convinced that `TEE_BigIntSizeInU32(bs1 + bs2)` is 100% correct. Perhaps `TEE_BigIntSizeInU32(ROUNDUP(bs1, 32) + ROUNDUP(b2, 32))`, but let's keep it simple.